bubuku/features/rebalance/change.py

```diff
@@ -0,0 +1,248 @@
+                        top_candidate = broker_pair
+            # Taking best topic to move
+            topic, _ = candidates[top_candidate][0]
+            # Updating cardinality map
+            for broker_pair, weight_list in candidates.items():
+                if top_candidate == broker_pair:
+                    OptimizedRebalanceChange._place_to_cardinalities(weight_list, topic, -2)
+                elif top_candidate[0] == broker_pair[0] or top_candidate[1] == broker_pair[1]:  # src or dst matches
+                    OptimizedRebalanceChange._place_to_cardinalities(weight_list, topic, -1)
+            # Get broker objects
+            source_broker, target_broker = brokers[top_candidate[0]], brokers[top_candidate[1]]
+
+            # Selecting best partition to move (It is better to just swap leadership, instead of copying data)
+            selected_partition = None
+            source_partitions = source_broker.list_partitions(topic, False)
+            target_partitions = target_broker.list_partitions(topic, True)
+            for partition in source_partitions:
+                if partition in target_partitions:
+                    selected_partition = partition
+                    break
+            # Failed to find optimized partition for swapping, taking any.
+            selected_partition = source_partitions[0] if selected_partition is None else selected_partition
+            topic_partition = (topic, selected_partition)
+            # Update brokers state
+            target_broker.accept_leader(source_broker, topic_partition)
+            # add transfer to queue
+            queue_list.append((topic_partition, source_broker.broker_id, target_broker.broker_id))
+            # Remove empty lists
+            for bp in [bp for bp, wl in candidates.items() if
+                       not brokers[bp[0]].have_extra_leaders() or not brokers[bp[1]].have_less_leaders()]:
+                del candidates[bp]
+        return queue_list
+
+    def _load_data(self):
+        self.broker_distribution = {id_: BrokerDescription(id_) for id_ in self.broker_ids}
+        self.source_distribution = {(topic, partition): replicas for topic, partition, replicas in
+                                    self.zk.load_partition_assignment()}
+        for topic_partition, replicas in self.source_distribution.items():
+            if not replicas:
+                continue
+            first = True
+            for replica in replicas:
+                if replica not in self.broker_distribution:
+                    self.broker_distribution[replica] = BrokerDescription(replica)
+                self.broker_distribution[replica].add_partition(first, topic_partition)
+                first = False
+        active_brokers = [broker for id_, broker in self.broker_distribution.items() if id_ in self.broker_ids]
+        distribute(
+            sum(b.get_leader_count() for b in self.broker_distribution.values()),
```

Comment:
I think that making extensive use of temporary variables could improve clarity. I would suggest to assign this expression to a variable like `total_leader_count`.